pkg/skaffold/test/custom/custom.go

```diff
@@ -32,34 +36,40 @@ import (
 	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
 )
 
-// for tests
-var doRunCustomCommand = runCustomCommand
+var (
+	// for tests
+	testContext = retrieveTestContext
+)
 
 const Windows string = "windows"
 
 type Runner struct {
 	customTest     latest.CustomTest
+	imageName      string
 	testWorkingDir string
 }
 
 // New creates a new custom.Runner.
-func New(cfg docker.Config, wd string, ct latest.CustomTest) (*Runner, error) {
+func New(cfg docker.Config, imageName string, wd string, ct latest.CustomTest) (*Runner, error) {
 	return &Runner{
+		imageName:      imageName,
 		customTest:     ct,
 		testWorkingDir: wd,
 	}, nil
 }
 
 // Test is the entrypoint for running custom tests
-func (ct *Runner) Test(ctx context.Context, out io.Writer, _ []build.Artifact) error {
-	if err := doRunCustomCommand(ctx, out, ct.customTest); err != nil {
+func (ct *Runner) Test(ctx context.Context, out io.Writer, artifacts []build.Artifact) error {
+	if err := ct.runCustomCommand(ctx, out, artifacts); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func runCustomCommand(ctx context.Context, out io.Writer, test latest.CustomTest) error {
+func (ct *Runner) runCustomCommand(ctx context.Context, out io.Writer, artifacts []build.Artifact) error {
```

Comment:
you can remove `runCustomCommand` function and move everything inside `Test` since it's identical to `Test` function and doesn't do anything extra.

Reply:
Merged `runCustomCommand()` with `Test()`.